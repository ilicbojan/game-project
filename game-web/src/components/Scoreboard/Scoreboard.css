.scoreboard {
  flex: 1;
  margin: 0;
}

.scoreboard-rounds {
  margin-bottom: 1rem;
}

.scoreboard-values {
  font-size: 1.5rem;
}

.scoreboard-wins-value {
  color: green;
}

.scoreboard-loses-value {
  color: red;
}

.scoreboard-ties-value {
  color: orange;
}

.scoreboard-wins-value,
.scoreboard-loses-value,
.scoreboard-ties-value {
  font-weight: bold;
}
