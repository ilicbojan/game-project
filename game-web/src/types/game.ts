export interface ChoiceItem {
  id: number;
  name: string;
}

export interface PlayRequest {
  player: number;
}

export interface PlayResponse {
  player: number;
  computer: number;
  result: ResultType;
}

export interface GameResult {
  player: string;
  computer: string;
  result: ResultType;
  date: Date;
}

export interface Score {
  win: number;
  lose: number;
  tie: number;
}

export enum ResultType {
  Win = "win",
  Lose = "lose",
  Tie = "tie",
}

export enum ChoiceId {
  Rock = 1,
  Paper = 2,
  Scissors = 3,
  Lizard = 4,
  Spock = 5,
}

export enum ChoiceName {
  Rock = "rock",
  Paper = "paper",
  Scissors = "scissors",
  Lizard = "lizard",
  Spock = "spock",
}
