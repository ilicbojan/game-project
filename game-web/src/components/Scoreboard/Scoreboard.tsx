import React from "react";
import { Score } from "../../types/game";
import "./Scoreboard.css";

interface Props {
  score: Score;
}

const Scoreboard: React.FC<Props> = ({ score }) => {
  return (
    <div className="scoreboard section">
      <h3>Scoreboard</h3>
      <div className="scoreboard-rounds">
        Rounds played: {score.win + score.lose + score.tie}
      </div>
      <div className="scoreboard-values">
        Wins: <span className="scoreboard-wins-value">{score.win}</span> |
        Losses: <span className="scoreboard-loses-value">{score.lose}</span> |
        Ties: <span className="scoreboard-ties-value">{score.tie}</span>
      </div>
    </div>
  );
};

export default Scoreboard;
