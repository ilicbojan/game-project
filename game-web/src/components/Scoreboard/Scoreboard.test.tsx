import { render, screen } from "@testing-library/react";
import { Score } from "../../types/game";
import Scoreboard from "./Scoreboard";

describe("Scoreboard", () => {
  const scoreMock: Score = {
    win: 3,
    lose: 1,
    tie: 2,
  };

  it("matches the snapshot", () => {
    const { asFragment } = render(<Scoreboard score={scoreMock} />);

    expect(asFragment()).toMatchSnapshot();
  });

  it("renders the score values correctly", () => {
    const { container } = render(<Scoreboard score={scoreMock} />);

    expect(screen.getByText("Scoreboard")).toBeInTheDocument();
    expect(screen.getByText(/Rounds played: 6/i)).toBeInTheDocument();

    expect(screen.getByText(/Wins:/)).toBeInTheDocument();
    expect(screen.getByText(/Losses:/)).toBeInTheDocument();
    expect(screen.getByText(/Ties:/)).toBeInTheDocument();

    expect(container.querySelector(".scoreboard-wins-value")?.textContent).toBe(
      "3"
    );
    expect(
      container.querySelector(".scoreboard-loses-value")?.textContent
    ).toBe("1");
    expect(container.querySelector(".scoreboard-ties-value")?.textContent).toBe(
      "2"
    );
  });
});
