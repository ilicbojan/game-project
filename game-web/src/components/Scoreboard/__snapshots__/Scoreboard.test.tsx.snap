// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Scoreboard matches the snapshot 1`] = `
<DocumentFragment>
  <div
    class="scoreboard section"
  >
    <h3>
      Scoreboard
    </h3>
    <div
      class="scoreboard-rounds"
    >
      Rounds played: 6
    </div>
    <div
      class="scoreboard-values"
    >
      Wins: 
      <span
        class="scoreboard-wins-value"
      >
        3
      </span>
       | Losses: 
      <span
        class="scoreboard-loses-value"
      >
        1
      </span>
       | Ties: 
      <span
        class="scoreboard-ties-value"
      >
        2
      </span>
    </div>
  </div>
</DocumentFragment>
`;
